### ISSUE TITLE
Output encoding issue of the Test Runner

### ISSUE THREAD

**Issue by sunnynie** [Microsoft/vscode-java-test#632]
Issue Type: <b>Bug</b>

Chinese is not supported,The output of English is acceptable.
    @Test
    public void test(){
        System.out.println("dd");
        System.out.println("等待");
    }

Extension version: 0.14.1
VS Code version: Code 1.31.1 (1b8e8302e405050205e69b59abb3559592bb9e60, 2019-02-12T02:20:54.427Z)
OS version: Windows_NT x64 10.0.17134

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i3-4160 CPU @ 3.60GHz (4 x 3592)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: enabled<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Memory (System)|7.87GB (2.02GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|

</details>

**Comment by sunnynie** [comment]
I hope it can be repaired as soon as possible. Thank you.

**Comment by yaohaizh** [comment]
@sunnynie Which part of Chinese is not display? The output? What is the version of VSCode Java redhat extension?

**Comment by Salandia** [comment]
me too
Language Support for Java(TM) by Red Hat redhat.java @0.40.0

**Comment by Salandia** [comment]
me too,like #652 
Language Support for Java(TM) by Red Hat redhat.java @0.40.0
Java Test Runner @0.15.0

**Comment by gculik** [comment]
Same here, the Java Test Runner Output is not handling Unicode properly, while the standard Output works fine.
`
    @Test
    public void testEncoding()
    {
        System.out.println("\u00B0C");
    }
`
outputs
`�C`

whereas
`
    public static void main(String[] args)
    {
        System.out.println("\u00B0C");
    }
`
correctly outputs
`°C`

**Comment by jdneo** [comment]
@all, the root cause of this issue is that:

When the Java language server compiling the source file, the file encoding will set to `utf-8` by default, or the user can customize the encoding with the setting `files.encoding`.

When running the test cases with the Test Runner, if the encoding setting is not equal with the files getting compiled, then those messy code will appear.

If you hasn't set the setting `files.encoding`, then that means the files are compiled with `utf-8` encoding. You can simply add the following test configurations to correct the encoding as a workaround:

```json
"java.test.config": {
    "vmargs": ["-Dfile.encoding=utf-8"]
}
```

I'll add some detecting logics later to better solve this problem.

**Comment by jdneo** [comment]
This issue should be fixed in #654.

So we have several ways to set the file encoding when running tests:

1. By default is `utf-8`.
2. The encoding set in the setting `files.encoding`
3. The encoding set in the test configuration. for example:
```json
"java.test.config": {
    "vmargs": ["-Dfile.encoding=Cp1252"]
},
```

The higher ones will be overridden by the lower ones.
